internal/jobs/tiktokapify/client.go

```diff
@@ -0,0 +1,97 @@
+package tiktokapify
+
+import (
+	"encoding/json"
+	"fmt"
+
+	teeargs "github.com/masa-finance/tee-types/args"
+	teetypes "github.com/masa-finance/tee-types/types"
+	"github.com/masa-finance/tee-worker/pkg/client"
+)
+
+const (
+	// Actors
+	SearchActorID   = "epctex~tiktok-search-scraper"                   // must rent this actor from apify explicitly
+	TrendingActorID = "lexis-solutions~tiktok-trending-videos-scraper" // must rent this actor from apify explicitly
+)
+
+type TikTokApifyClient struct {
+	apify *client.ApifyClient
+}
+
+func NewTikTokApifyClient(apiToken string) (*TikTokApifyClient, error) {
+	apifyClient, err := client.NewApifyClient(apiToken)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Apify client: %w", err)
+	}
+	return &TikTokApifyClient{apify: apifyClient}, nil
+}
+
+// ValidateApiKey validates the underlying Apify API token
+func (c *TikTokApifyClient) ValidateApiKey() error {
+	return c.apify.ValidateApiKey()
+}
+
+// SearchByQuery runs the search actor and returns typed results
+func (c *TikTokApifyClient) SearchByQuery(input teeargs.TikTokSearchByQueryArguments, cursor client.Cursor, limit int) ([]*teetypes.TikTokSearchByQueryResult, client.Cursor, error) {
+	// Map snake_case fields to Apify actor's expected camelCase input
+	startUrls := input.StartUrls
+	if startUrls == nil {
+		startUrls = []string{}
+	}
+	searchTerms := input.Search
+	if searchTerms == nil {
+		searchTerms = []string{}
+	}
+
+	apifyInput := map[string]any{
+		"search":    searchTerms,
+		"startUrls": startUrls,
+		"maxItems":  input.MaxItems,
+		"endPage":   input.EndPage,
+	}
```

Comment:
I'm wondering which one makes more sense to talk with Apify, `map[string]any` or use an actual `struct` (e.g. https://github.com/masa-finance/tee-worker/blob/main/internal/jobs/redditapify/client.go#L56). At least in Reddit it probably makes sense to use a struct since it uses the same structure for all the query types, but not sure how generally applicable it is.

Reply:
fair point - converted both types to request structs for consistency!